**Q1**

H0 = There is no significant difference in diameter of the cutlet

HA = There is significant difference in the diameter of the cutlet 

In [ ]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [ ]:
# Load the dataset
data=pd.read_csv('/content/Cutlets.csv')
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
data.shape

(35, 2)

In [ ]:
data.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [ ]:
stats.ttest_ind(data['Unit A'],data['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

p-value = 0.472

with 5% significance level, the Alpha value is 0.05

p-value > Alpha [0.47 > 0.05]

hence Fail to reject Null Hypothesis[H0]

There is no significant difference in diameter of the cutlet


**Q2 LabTat**

H0 = No difference in average TAT among the different laboratories

HA = Difference in average TAT among the different laboratories

In [ ]:
# Load dataset

data = pd.read_csv('/content/LabTAT.csv')


In [ ]:
# descriptive statistics

data.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [ ]:
stats.f_oneway(data.iloc[:,0],data.iloc[:,1],data.iloc[:,2],data.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)



*   p-value ~ 0
*   with 5% significance level, the alpha value is 0.05

*   p-value < alpha [~ 0 < 0.05]
*   Reject [H0] Null hypothesis





There is difference in average TAT among different laboratories

**Q3** **Buyer ratio**

H0 = All proportion are equal

HA = Not all proportions are equal

In [ ]:
import numpy as np
from scipy.stats import chi2_contingency

In [ ]:
# Load the data

data = pd.read_csv('/content/BuyerRatio.csv')

In [ ]:
# descriptive statistic

data.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [ ]:
data

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
observed_value = np.array([[50,142,131,70],[435,1523,1356,750]])

In [ ]:
p_value = chi2_contingency(observed_value)   # here output is chi2 stats value,p_value, degrees of freedom,expected values
p_value


(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [ ]:
p_value[1]

0.6603094907091882



*   p-value = 0.660
*   with 5% significance level, the alpha value is 0.05
*   p_value > alpha [0.660 > 0.05]
*   fail to reject null hypothesis [H0]



Not all proportions are equal





**Q4 customer order form**

H0 = No defective varies by centre

HA = Defective varies by centre

In [ ]:
# load dataset

data = pd.read_csv('/content/Costomer+OrderForm.csv')

In [ ]:
data.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [ ]:
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [ ]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [ ]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [ ]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [ ]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [ ]:
observed_value = np.array([
                   [271,267,269,280],
                  [29,33,31,20]
            ])

In [ ]:
p_value = chi2_contingency(observed_value)
p_value

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [ ]:
p_value[1]

0.2771020991233135



*   p-value = 0.277
*   with 5% significance level, the alpha value is 0.05


*   p-value > alpha [0.277 > 0.05]
*   fail to reject null hypothesis [H0]


defects are vary by center


